RockLib.Immutable.Tests/SemimutableTests.cs

```diff
@@ -3,160 +3,160 @@
+         semimutable.Value = 2;
 
-            semimutable.ResetValue();
+         semimutable.ResetValue();
 
-            Assert.True(semimutable.HasDefaultValue);
-        }
+         Assert.True(semimutable.HasDefaultValue);
+      }
 
-        [Fact]
-        public void HasDefaultValueIsTrueWhenValueIsNotChanged()
-        {
-            var semimutable = new Semimutable<int>(1);
+      [Fact]
+      public void HasDefaultValueIsTrueWhenValueIsNotChanged()
+      {
+         var semimutable = new Semimutable<int>(1);
 
-            Assert.True(semimutable.HasDefaultValue);
-        }
+         Assert.True(semimutable.HasDefaultValue);
+      }
 
-        [Fact]
-        public void HasDefaultValueIsFalseWhenValueIsChanged()
-        {
-            var semimutable = new Semimutable<int>(1);
+      [Fact]
+      public void HasDefaultValueIsFalseWhenValueIsChanged()
+      {
+         var semimutable = new Semimutable<int>(1);
 
-            semimutable.Value = 2;
+         semimutable.Value = 2;
 
-            Assert.False(semimutable.HasDefaultValue);
-        }
+         Assert.False(semimutable.HasDefaultValue);
+      }
 
-        [Fact]
-        public void HasDefaultValueIsFalseWhenSetValueIsCalled()
-        {
-            var semimutable = new Semimutable<int>(1);
+      [Fact]
+      public void HasDefaultValueIsFalseWhenSetValueIsCalled()
+      {
+         var semimutable = new Semimutable<int>(1);
 
-            semimutable.SetValue(() => 2);
+         semimutable.SetValue(() => 2);
 
-            Assert.False(semimutable.HasDefaultValue);
-        }
+         Assert.False(semimutable.HasDefaultValue);
+      }
 
-        [Fact]
-        public void HasDefaultValueIsTrueWhenResetValueIsCalled()
-        {
-            var semimutable = new Semimutable<int>(1);
+      [Fact]
+      public void HasDefaultValueIsTrueWhenResetValueIsCalled()
+      {
+         var semimutable = new Semimutable<int>(1);
 
-            semimutable.Value = 2;
-            semimutable.ResetValue();
+         semimutable.Value = 2;
+         semimutable.ResetValue();
 
-            Assert.True(semimutable.HasDefaultValue);
-        }
+         Assert.True(semimutable.HasDefaultValue);
+      }
 
-        [Fact]
-        public void IsLockedIsFalseInitially()
-        {
-            var semimutable = new Semimutable<int>(1);
+      [Fact]
+      public void IsLockedIsFalseInitially()
+      {
+         var semimutable = new Semimutable<int>(1);
 
-            Assert.False(semimutable.IsLocked);
-        }
+         Assert.False(semimutable.IsLocked);
+      }
 
-        [Fact]
-        public void IsLockedIsTrueAfterTheValuePropertyIsAccessed()
-        {
-            var semimutable = new Semimutable<int>(1);
+      [Fact]
+      public void IsLockedIsTrueAfterTheValuePropertyIsAccessed()
+      {
+         var semimutable = new Semimutable<int>(1);
 
-            var value = semimutable.Value;
+         _ = semimutable.Value;
```

Comment:
Well that's cool.  I haven't seen this before so I had to look it up.  I like it. 👍🏻

Reply:
It gets around an analyzer that says "you've assigned a value but you're not using it". I think there's another one that says you shouldn't ignore return values. The discard variable is nice because it addresses both of them: I'm assigning it, but I explicitly don't care about it :).

https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/discards